src/bot.py

```diff
@@ -188,14 +193,33 @@ def on_maintenance(self):
         # Check if merc needs to be revived
         merc_alive, _ = self._template_finder.search("MERC", self._screen.grab(), threshold=0.9, roi=[0, 0, 200, 200])
         if not merc_alive:
-            Logger.info("Reviveing merc")
+            Logger.info("Reviving merc")
             if not self._pather.traverse_nodes(self._curr_location, Location.QUAL_KEHK, self._char):
                 self.trigger("end_game")
                 return
             self._curr_location = Location.QUAL_KEHK
             if self._npc_manager.open_npc_menu(Npc.QUAL_KEHK):
                 self._npc_manager.press_npc_btn(Npc.QUAL_KEHK, "resurrect")
             time.sleep(2)
+        merc_alive, _ = self._template_finder.search("MERC", self._screen.grab(), threshold=0.9, roi=[0, 0, 200, 200])
+        if merc_alive:
+            self._merc_revive_attempts = 0
+        elif self._config.general["require_merc"]:
+            self._merc_revive_attempts = self._merc_revive_attempts + 1
+            mouse.move(0,0, randomize=[5, 5], delay_factor=[0.5,1])
+            wait(0.2, 0.3)
+            mouse.click(button="left")
+            wait(0.3, 0.5)
+            mouse.click(button="left")
+            wait(0.5, 1)
```

Comment:
What is this supposed to do?

Reply:
@aeon0 For one, sometimes resurrects are not performed because the NPC can't be located and times out, so then character will attempt run. If that case happens, the way you have written it will cause it to close out rather than try one more time.

The reason for that particular code block is that if you don't have enough gold to resurrect the merc, you need to click away twice to close the "not enough gold" dialogue.

Reply:
But doesnt it then vialoate the "require_merc" when you run once without. Seems a little hacky to me to say, "ok we try it one more time next round". And bloats up the code.
What about increasing the time_out to look for Qual if require_merc = True?

Also side note. Lets try not to have any implementation details in bot.py. It already gets quite big. A good indicator that code should not be inside bot.py is if it uses the mouse and keyboard module.

Reply:
I've watched my bot endlessly chase NPC's maybe twice before ultimately timing out and my mercless bot runs runs into Pindle's warm embrace, so that's why I implemented it that way. :P But hopefully more time should be sufficient.